sdk/mobile-center/src/main/java/com/microsoft/azure/mobile/channel/DefaultChannel.java

```diff
@@ -384,22 +385,50 @@ private synchronized void triggerIngestion(final String batchId, final GroupStat
             /* Remember this batch. */
             groupState.mSendingBatches.put(batchId, batch);
 
+            /*
+             * Due to bug on old Android versions (verified on 4.0.4),
+             * if we start an async task from here, i.e. the async persistence handler thread,
+             * we end up with AsyncTask configured with the wrong Handler to use for onPostExecute
+             * instead of using main thread as advertised in Javadoc (and its a static field there).
+             *
+             * Our SDK guards against an application that would make a first async task in non UI
+             * thread before SDK is initialized, but we should also avoid corrupting AsyncTask
+             * with our wrong handler to avoid creating bugs in the application code since we are
+             * a library.
+             *
+             * So make sure we execute the async task from UI thread to avoid any issue.
+             */
+            HandlerUtils.runOnUiThread(new Runnable() {
```

Comment:
Because as I explained in the comment, the first async task created anywhere in the application will trigger the creation of the static Handler used by AsyncTask implementation.

On API level 15 (at least, and I know in 19 it's fixed, maybe sooner), they create the Handler using current thread instead of specifying main looper. As a result, the onPostExecute method will be in the wrong thread as the Handler used to call onPostExecute is not initialized correctly, I sent you on slack the code snippet that shows the Android bug.